File: ss.pl
use strict;
use warnings;

my $c = 0;
my $d = 0;
open(IN, "a.txt");
  foreach ( <IN> ){
    $d++;
    if( $_ =~ /EMP2017/gi){
	  $c = -1;
	}
	if( $_=~ /RRD2017/gi){
	  $c++;
	}
	if( $c == 1 ){
	  print "$d : $c\n";
	}
  }
close(IN);

File: ss.py
import re

c = 0
d = 0
e = 0

lines = open("a.txt", "r")
for i in lines.readlines():
  d += 1
  if( re.search(r'EMP2017', i)):
    c = -1
    e = d
  if( re.search(r'RRD2017', i)):
    c += 1
  if( c == 1 ):
    print (e , " : ", d)
lines.close()